validator/validator.go

```diff
@@ -0,0 +1,61 @@
+// Package validator usage:
```

Comment:
This makes the file too verbose, can we use `doc.go` instead for documentation purpose?